### PR TITLE
Adds an dart-synthesizing dartgun for CMO/Chemist traitors

### PR DESCRIPTION
**Pull request by astralloop**
:cl: Astral
add: Traitor CMOs and Chemists, for 12 TC, can now get a reagent dartgun, which is capable of synthesizing it's own syringes, but does so slowly, and can be easily identified as syndicate by anyone who isn't blind!
/:cl:

 
Reagent Dartgun: left, Syringe Gun: right

### CODE REVIEW

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -1275,6 +1275,13 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	cost = 14
 	restricted_roles = list("Geneticist", "Chief Medical Officer")
 
+/datum/uplink_item/role_restricted/chemical_gun
+	name = "Reagent Dartgun"
+	desc = "A heavily modified syringe gun, which is capable of synthesizing it's own chemical darts, using input reagents."
```
Comment (PKPenguin321) [inline code/modules/uplink/uplink_items.dm]: No commas are needed in this sentence

Comment (astralloop) [reply code/modules/uplink/uplink_items.dm]: Fixed

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1275,6 +1275,13 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	cost = 14
 	restricted_roles = list("Geneticist", "Chief Medical Officer")
 
+/datum/uplink_item/role_restricted/chemical_gun
+	name = "Reagent Dartgun"
+	desc = "A heavily modified syringe gun which is capable of synthesizing it's own chemical darts using input reagents. Can hold 100u of reagents."
```
Comment (QualityVan) [inline code/modules/uplink/uplink_items.dm]: its

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -1277,7 +1277,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 
 /datum/uplink_item/role_restricted/chemical_gun
 	name = "Reagent Dartgun"
-	desc = "A heavily modified syringe gun, which is capable of synthesizing it's own chemical darts, using input reagents."
+	desc = "A heavily modified syringe gun which is capable of synthesizing it's own chemical darts using input reagents. Can hold 100u of reagents."
```
Comment (PKPenguin321) [inline code/modules/uplink/uplink_items.dm]: Oh and I don't think `it's` needs an apostrophy here

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (PKPenguin321) [reply code/modules/uplink/uplink_items.dm]: fug

**Comment by epochayur** [comment]
mechanically, how is this different from a syringe gun?

**Comment by ExcessiveUseOfCobblestone** [comment]
A hardtell at the cost of getting the non-limiting aspect of a syringe gun ???

**Comment by astralloop** [comment]
@epochayur @ExcessiveUseOfCobblestone 

you don't have to drag around 2 boxes of syringes and refill them and put them in  as you go, but at the cost of being super freaking obvious (red sprite / red inhand / obvious name on examine) and also if you use them all, you have to wait for the 25 sec recharge

**Comment by Robustin** [comment]
Aside from making it maybe 12TC I'd say this is good. People are underestimating it since having to stop and manually load syringes is the single greatest downside of syringe guns.

**Comment by astralloop** [comment]
it's 12TC now

**Comment by MoreRobustThanYou** [comment]
hey this reminds me of goon, nice

**Comment by ExcessiveUseOfCobblestone** [comment]
@Robustin but it doesn't synth the chem so you have to load it anyways...

**Comment by Wilchenx** [comment]
>buffing chemists

**Comment by evsey9** [comment]
wait it doesnt synth the chem?????
why would anyone ever use this

**Comment by Wilchenx** [comment]
so how it works it synthesizes a syringe


that you have to take out


and fill with chems


then put in back in again?

**Comment by kevinz000** [comment]
:-1:, the limited availability of syringes even with autolathes being prevalent is a balancing factor to one hit kills via killmixes. while you can only carry so many syringes (49 if you use your entire backpack and carry a syringe gun in hand), you can carry a shitload more reagents (a large beaker takes the same amount of space and can fill 100/15 = 6.33 syringes).

**Comment by XDTM** [comment]
To those confused: you can pour up to 100u of reagents into the gun, which end up in the synthesized syringes.

**Comment by evsey9** [comment]
wait, you can make piercing syringes? how?

**Comment by ExcessiveUseOfCobblestone** [comment]
R&D and Medical Proto

**Comment by evsey9** [comment]
what research?

**Comment by astralloop** [comment]
no, you can't upgrade the syringes in this, they'll always be normal, non-cryo, non-piercing, non-high capacity darts

**Comment by astralloop** [comment]
@kevinz000 also, if you spam-click this, you'll get an initial burst of 60u, but after, if you continue spamclicking constantly, around 0.6u per second (One 15u syringe per 25 seconds)

**Comment by Xhuis** [comment]
It seems kind of strange since if someone sees it they can just... *dart* away!

**Comment by ExcessiveUseOfCobblestone** [comment]
@evsey9 [advanced biotech](https://github.com/tgstation/tgstation/blob/78fc25e2a02015e41b3f55648aa69a830778cca3/code/modules/research/techweb/all_nodes.dm#L24)

**Comment by vuonojenmustaturska** [comment]
fuck it, let's try it

**Pull request merged**